### PR TITLE
Tests for cmap and imap files

### PR DESCRIPTION
**Pull request by rajul**
Added tests for the cmap and imap files

### CODE REVIEW

File: ginga/tests/test_cmap.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        test_clst = ((0.0, 0.0, 0.0), (1.0, 1.0, 1.0))
+        test_color_map = ColorMap('test-name', test_clst)
+
+        expected = 'test-name'
+        actual = test_color_map.name
+        assert expected == actual
+
+        expected = 2
+        actual = len(test_color_map.clst)
+        assert expected == actual
```
Comment (ejeschke) [inline ginga/tests/test_cmap.py]: Hi Rajul,

Thanks for making these tests!  ginga color maps have to be of length 256.  It is possible to stretch ones that are smaller to meet this restriction, as is done with the matplotlib colormaps.  But I think a test should create one of the correct size.  It's pretty easy to use `np.linspace()` (for example) to create arrays of the correct length.

File: ginga/tests/test_imap.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        test_ilst = (0.0, 1.0, 1.0, 1.0)
+        test_intensity_map = IntensityMap('test-name', test_ilst)
+
+        expected = 'test-name'
+        actual = test_intensity_map.name
+        assert expected == actual
+
+        expected = 4
+        actual = len(test_intensity_map.ilst)
+        assert expected == actual
```
Comment (ejeschke) [inline ginga/tests/test_imap.py]: Same thing for intensity maps.

**Comment by ejeschke** [comment]
I've added a couple of checks in `ginga.cmap` and `ginga.imap`.  You can use the constants `cmap.min_cmap_size` and `imap.min_imap_size` as the size they should be.

**Comment by rajul** [comment]
@ejeschke Eric, I suppose this can be merged. Are there any changes that you wish to be done here?

File: ginga/tests/test_cmap.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    def test_ColorMap_init(self):
+        test_clst = tuple([(x, x, x)
+                           for x in np.linspace(0, 1, ginga.cmap.min_cmap_len)])
+        test_color_map = ColorMap('test-name', test_clst)
+
+        expected = 'test-name'
+        actual = test_color_map.name
+        assert expected == actual
+
+        expected = 256
```
Comment (ejeschke) [inline ginga/tests/test_cmap.py]: @rajul, can you change this from "256" to `ginga.cmap.min_cmap_len`? This ill test that it is the correct length even if the size changes under the hood.

File: ginga/tests/test_imap.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+    def test_IntensityMap_init(self):
+        test_ilst = tuple(np.linspace(0, 1, ginga.imap.min_imap_len))
+        test_intensity_map = IntensityMap('test-name', test_ilst)
+
+        expected = 'test-name'
+        actual = test_intensity_map.name
+        assert expected == actual
+
+        expected = 256
```
Comment (ejeschke) [inline ginga/tests/test_imap.py]: Similarly, 256 --> `ginga.imap.min_imap_len`

**Comment by ejeschke** [comment]
@rajul, just a couple of last nit-picky changes, and I'll merge.

**Comment by ejeschke** [comment]
@rajul, if you want I can make these changes as I merge the PR. LMK.

**Comment by rajul** [comment]
@ejeschke Thanks Eric, I have made the changes as you suggested.

**Comment by ejeschke** [comment]
I'm not so sure it is a good idea to test the equality of the internal `clst` to the input.  The representation inside the object may change. But I guess we can change the test if that happens.

**Pull request merged**